lib/src/git.rs

```diff
@@ -1229,6 +1229,146 @@ pub enum GitFetchError {
+        // local branches.
+        let refspecs: Vec<_> = branch_names
+            .iter()
+            .map(|pattern| {
+                pattern
+                    .to_glob()
+                    .filter(
+                        /* This triggered by non-glob `*`s in addition to INVALID_REFSPEC_CHARS
+                         * because `to_glob()` escapes such `*`s as `[*]`. */
+                        |glob| !glob.contains(INVALID_REFSPEC_CHARS),
+                    )
+                    .map(|glob| format!("+refs/heads/{glob}:refs/remotes/{remote_name}/{glob}"))
+            })
+            .collect::<Option<_>>()
+            .ok_or(GitFetchError::InvalidBranchPattern)?;
+        if refspecs.is_empty() {
+            // Don't fall back to the base refspecs.
+            return Ok(None);
+        }
+
+        tracing::debug!("remote.download");
+        remote.download(&refspecs, Some(&mut self.fetch_options))?;
+        tracing::debug!("remote.prune");
+        remote.prune(None)?;
+        tracing::debug!("remote.update_tips");
+        remote.update_tips(
+            None,
+            git2::RemoteUpdateFlags::empty(),
+            git2::AutotagOption::Unspecified,
+            None,
+        )?;
+
+        self.fetched.push(FetchedBranches {
+            branches: branch_names.to_vec(),
+            remote: remote_name.to_string(),
+        });
+
+        let mut default_branch = None;
+        if let Ok(default_ref_buf) = remote.default_branch() {
+            if let Some(default_ref) = default_ref_buf.as_str() {
+                // LocalBranch here is the local branch on the remote, so it's really the remote
+                // branch
+                if let Some(RefName::LocalBranch(branch_name)) = parse_git_ref(default_ref) {
+                    tracing::debug!(default_branch = branch_name);
+                    default_branch = Some(branch_name);
+                }
+            }
+        }
+        tracing::debug!("remote.disconnect");
+        remote.disconnect()?;
+        Ok(default_branch)
+    }
+
+    pub fn import_refs(&mut self) -> Result<GitImportStats, GitImportError> {
+        // Import the remote-tracking branches into the jj repo and update jj's
+        // local branches. We also import local tags since remote tags should have
+        // been merged by Git.
+        tracing::debug!("import_refs");
+        let import_stats = import_some_refs(self.mut_repo, self.git_settings, |ref_name| {
+            self.fetched
+                .iter()
+                .filter_map(|fetched| {
+                    to_remote_branch(ref_name, &fetched.remote).map(|branch| {
+                        fetched
+                            .branches
+                            .iter()
+                            .any(|pattern| pattern.matches(branch))
+                    })
+                })
+                .next()
+                .unwrap_or(matches!(ref_name, RefName::Tag(_)))
```

Comment:
nit: It's probably better to do `match ref_name` first (instead of using `to_remote_branch()`. Something like:
```rust
match ref_name {
    RefName::LocalBranch(_) => false,
    RefName::RemoteBranch { branch, remote } => {
        self.fetched.get(remote).is_some_and(|patterns| ...)
    }
    RefName::Tag(_) => true,
}
```